compiler/rustc_codegen_llvm/src/debuginfo/mod.rs

```diff
@@ -224,9 +225,9 @@ pub struct DebugLoc {
     /// Information about the original source file.
     pub file: Lrc<SourceFile>,
     /// The (1-based) line number.
-    pub line: Option<u32>,
+    pub line: Option<NonZeroU32>,
```

Comment:
This is a rather convoluted way to describe a u32, especially because `UNKNOWN_LINE_NUMBER` is 0. Is this additional level of abstraction necessary here?

Reply:
In retrospect, since there's only a handful of places where `line` and `col` are used, a plain `u32` is probably a better fit. Also, `SourceInfo` in `mod.rs` already uses a `u32` over an `Option<u32>` to encode the same information, so I think this change is warranted.